### PR TITLE
Allow anon login

### PR DESCRIPTION
**Pull request by craigmulligan**
connects to #12

Change-Type: minor

### CODE REVIEW

**Comment by craigmulligan** [comment]
Hey @emirotin @pimterry, Do you mind having a quick look at this? It relaxes the authentication requirements just for gosquared. 

If you are fine with this approach I'll do the same for the rest of the services.

File: packages/resin-event-log/src/resin-event-log.js
Comment (pimterry) [inline packages/resin-event-log/src/resin-event-log.js:67]: This shouldn't go away totally, it should just move into the mixpanel/ga adapters, so those fail individually instead. That way this will be usable if you only have goSquared configured, and we can look at how to transparently support mixpanel & go squared at the same time separately later.

Right now, if you do try to call .login on those adapters without a user, as this PR does, I suspect you'll break things... Mixpanel for example will define & alias for a non-existent user id, which probably break all sorts. Testing welcome :smiley:

**Review by pimterry — CHANGES_REQUESTED** [review]

File: packages/resin-event-log/src/resin-event-log.js
Comment (craigmulligan) [reply packages/resin-event-log/src/resin-event-log.js:67]: @pimterry the idea was to add anon logins for all services with this PR, but I first want to do the one service and get feedback before tackling the others. If the approach is fine I'll do the remainder of the services (with tests).

Comment (pimterry) [reply packages/resin-event-log/src/resin-event-log.js:67]: Ok, sure. In that case, this part of the approach looks broadly fine to me, but it needs extending to the rest, and checking that that doesn't break anything, before we can merge it.

File: packages/resin-event-log/test/resin-event-log.spec.js
Comment (craigmulligan) [inline packages/resin-event-log/test/resin-event-log.spec.js]: @pimterry quick question here ^

Comment (pimterry) [reply packages/resin-event-log/test/resin-event-log.spec.js]: Because that's what we've historically been using as the distinct id everywhere. I don't think it matters which we use, since they're both unique identifiers, but it does need to be consistent everywhere, or we need to make sure we always alias them together. Easier to be consistent.

That's the historic position. For more current context, we're now (as of the UI deploy early this week I think), aliasing usernames to UUIDs at signup in the UI, so most users will really have UUID distinct ids. You can still track events and profile data against their usernames everywhere, but all that data will end up associated to the UUID profile instead.

This won't be 100% of users though, as we don't have proper handling for social signups (so they'll still just be usernames, and don't yet have funnel data), and it is possible to sign up without going through the UI at all, using the CLI or the API directly, where obviously we can't get funnel data anyways. Generally, using the username as a distinct id should end up doing the right thing everywhere.

Comment (emirotin) [reply packages/resin-event-log/test/resin-event-log.spec.js]: The problem with usernames is people want them to become mutable

**Comment by craigmulligan** [comment]
@pimterry @emirotin I was figuring out the current user identification behaviour of the module(with this PR), and I came up with this. 
Could you check if for any inaccuracies? Once we nailed down the current state we can talk about the target. 

```
browser: 

Anon Login: 
mixpanel, GA, gosquared -> all automatically generate a random “clientId/distinctId” (whatever you want to call it)

User Login: 
mixpanel  -> makes the distinct_id = user.username also aliases itself? 
GA -> makes uid = user.id (still generates a unique clientId)
Gosquared -> makes person.id = user.id (still generates unique clientID) 

Node: 

Anon Login: 
mixpanel ->  events are sent without any user/clientId attached.
GA -> generates a clientId everytime it’s run and attaches to events.
Gosquared -> events are sent without any user/clientId attached.

User Login: 
mixpanel  -> same as browser
GA -> same as browser 
Gosquared -> makes person.id = user.id (no clientId)
```

**Comment by emirotin** [comment]
> mixpanel  -> makes the distinct_id = user.username also aliases itself?

What it actually does is it links the username to the previously random generated ID and now all the ytracks mentioning the username are linked to that original random ID.

**Comment by craigmulligan** [comment]
Okay @emirotin @pimterry, 

I've added the functionality as described in the note above ^, also unified the test functions a little so you can optionally validate user/events just by passing them in the createMock(options) instead of rewriting the queryFilter function every time (it's how we were doing it for mixpanel).

cc @izavits you may want to take a look at this too.

File: packages/resin-event-log/src/adaptors/mixpanel.js
Comment (emirotin) [inline packages/resin-event-log/src/adaptors/mixpanel.js]: A better style and the more minimal change would be to lift the `if (!user) {}` to the top of the method and keep the rest intact

Comment (craigmulligan) [reply packages/resin-event-log/src/adaptors/mixpanel.js]: Ah yes, much better!

**Comment by craigmulligan** [comment]
@pimterry @emirotin could you take another look?

File: packages/resin-event-log/src/resin-event-log.js
Comment (emirotin) [inline packages/resin-event-log/src/resin-event-log.js]: Could just use simple `!user.id` checks?

File: packages/resin-universal-ga/src/browser.js
Comment (emirotin) [inline packages/resin-universal-ga/src/browser.js:7]: Oh was it an accidental global? :(

**Comment by emirotin** [comment]
LGTM in general, please wait for Tim's review too

**Review by pimterry — APPROVED** [review]

**Comment by craigmulligan** [comment]
Sweet, thanks for the review chaps. Looks like update from master broke the tests, will need to investigate.

**Comment by craigmulligan** [comment]
@emirotin fixed the tests, but strangely some of the GA tests are now failing locally for me, (even on a fresh clone on `master`), must be some environmental issue? Can you pull master and run the tests to check? It may just be my machine for some reason.

**Comment by emirotin** [comment]
Tests work for me, and on your brunch too

**Comment by craigmulligan** [comment]
My brunch too 🍳? Cool, must be something will weird with my machine, I'll squash n merge.

**Pull request merged**
